src/api/api_models.rs

```diff
@@ -197,9 +197,14 @@ pub struct PlaylistOwner {
     pub display_name: String,
 }
 
+const def_image: &'static [Image] = &[Image {url: String::new(), height: Some(640), width: Some(640)}];
+
 impl WithImages for Playlist {
     fn images(&self) -> &[Image] {
-        &self.images[..]
+        match &self.images {
+            Some(x) => &x[..],
+            None => &def_image[..],
```

Comment:
Can't we return an empty array here?

Reply:
No. Rust returns `returns a value referencing data owned by the current function`. I believe it has to do with garbage collection, but I do not know much rust so I just got around the issue by making a static constant outside the implementation.